### PR TITLE
doc: how F3 sidecar interact with forest

### PR DESCRIPTION
**Pull request by hanabi1224**
## Summary of changes

This PR adds a mermaid diagram on how F3 sidecar interacts with Forest.

To preview the diagram: https://github.com/ChainSafe/forest/blob/hm/f3-doc/f3-sidecar/README.md

Changes introduced in this pull request:

-

## Reference issue to close (if applicable)

Closes

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

File: f3-sidecar/README.md [hunk 1]
```diff
@@ -12,3 +12,16 @@
 - (optional) to inspect RPC calls, run
   `mitmproxy --mode reverse:http://localhost:2345 --listen-port 8080` then
   `go run . -rpc http://127.0.0.1:8080/rpc/v1`
+
+### How F3 sidecar interacts with Forest
+
+```mermaid
+flowchart TD
+    A[F3 sidecar] -->|EC API calls| B(Forest)
```
[the remark this answers is not in the document]
Comment (hanabi1224) [reply f3-sidecar/README.md:42]: Descriptions updated.

File: f3-sidecar/README.md [hunk 2]
```diff
@@ -12,3 +12,38 @@
 - (optional) to inspect RPC calls, run
   `mitmproxy --mode reverse:http://localhost:2345 --listen-port 8080` then
   `go run . -rpc http://127.0.0.1:8080/rpc/v1`
+
+### How F3 sidecar interacts with Forest
+
+An F3 sidecar node is a standalone node that is part of a p2p network and
```
Comment (ruseinov) [inline f3-sidecar/README.md:18]:
```suggestion
An F3 sidecar node is a standalone node that is a part of a p2p network and
```

File: f3-sidecar/README.md [hunk 3]
```diff
@@ -12,3 +12,38 @@
...
+participates in the f3 protocol.
+
+Besides what have been handled internally(e.g. p2p communications) in the
+`go-f3` lib
+
+- it uses `level-db` as storage backend
```
Comment (ruseinov) [inline f3-sidecar/README.md:24]:
```suggestion
- it uses `level-db` as a storage backend
```

**Review by LesnyRumcajs — APPROVED** [review]
rock-solid

**Review by ruseinov — APPROVED** [review]

**Pull request merged**